saga/saga_state.go

```diff
@@ -244,6 +244,23 @@ func (state *SagaState) addTaskData(taskId string, msgType SagaMessageType, data
 	}
 }
 
+/*
+ * Applies the supplied messages atomically to the supplied sagaState.
+ * Returns the mutated sate if all the messages are valid,
+ * otherwise returns the original state if any of the messages are invalid along with the error
+ */
+func bulkUpdateSagaState(state *SagaState, msgs []SagaMessage) (*SagaState, error) {
+	originalState := copySagaState(state)
```

Comment:
Agree, a thing to note is that this case(updates with bulk messages) does not occur so often(more info in the above comment). If we don't save the old state and error occurs, we'll have to error the entire batch of updates(upto 10,000 in this case), since the current state would be unusable.